src/components/menu/main-menu.tsx

```diff
@@ -4,81 +4,155 @@ import Link from "@components/elements/link";
+import {
+  useCallback,
+  useEffect,
+  useId,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { usePathname } from "next/navigation";
 
 const menuLevelsToShow = 2;
 
 type Props = {
   /**
    * Array of nested menu items.
    */
-  menuItems: MenuItemType[]
-}
+  menuItems: MenuItemType[];
+};
 
-const MainMenu = ({menuItems}: Props) => {
-  const buttonRef = useRef<HTMLButtonElement>(null)
+const MainMenu = ({ menuItems }: Props) => {
+  const buttonRef = useRef<HTMLButtonElement>(null);
   const menuRef = useRef<HTMLDivElement>(null);
   const navId = useId();
 
-  const {value: menuOpen, setFalse: closeMenu, toggle: toggleMenu} = useBoolean(false)
-  const browserUrl = usePathname()
+  const {
+    value: menuOpen,
+    setFalse: closeMenu,
+    toggle: toggleMenu,
+  } = useBoolean(false);
+  const browserUrl = usePathname();
   const activeTrail = useActiveTrail(menuItems, usePathname() || "");
 
   useOutsideClick(menuRef, closeMenu);
 
-  const handleEscape = useCallback((event: KeyboardEvent) => {
-    if (event.key !== "Escape" || !menuOpen) return;
+  const handleEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key !== "Escape" || !menuOpen) return;
 
-    closeMenu()
-    buttonRef.current?.focus();
-  }, [menuOpen, closeMenu]);
+      closeMenu();
+      buttonRef.current?.focus();
+    },
+    [menuOpen, closeMenu]
+  );
 
   useEffect(() => closeMenu(), [browserUrl, closeMenu]);
   useEventListener("keydown", handleEscape, menuRef);
 
   return (
-    <nav id={navId} aria-label="Main Navigation" className="lg:centered" ref={menuRef}>
-      <button ref={buttonRef} className="flex flex-col items-center lg:hidden absolute top-5 right-10 group" onClick={toggleMenu} aria-expanded={menuOpen} aria-labelledby={navId}>
+    <nav
+      id={navId}
+      aria-label="Main Navigation"
+      className="lg:centered"
+      ref={menuRef}
+    >
+      <button
+        ref={buttonRef}
+        className="flex flex-col items-center lg:hidden absolute top-5 right-10 rs-py-2 group"
+        onClick={toggleMenu}
+        aria-expanded={menuOpen}
+        aria-labelledby={navId}
+      >
         <span className="flex flex-col justify-center items-center w-[30px] h-[30px]">
-          <span className={clsx("bg-black-true block transition-all duration-300 ease-out h-[3px] w-full rounded-sm", {"rotate-45 translate-y-4": menuOpen, "-translate-y-0.5": !menuOpen})}/>
-          <span className={clsx("bg-black-true block transition-all duration-300 ease-out h-[3px] w-full rounded-sm my-3",{"opacity-0" :menuOpen , "opacity-100": !menuOpen})}/>
-          <span className={clsx("bg-black-true block transition-all duration-300 ease-out h-[3px] w-full rounded-sm", {"-rotate-45 -translate-y-4": menuOpen, "translate-y-0.5": !menuOpen})}/>
+          <span
+            className={clsx(
+              "bg-black-true block transition-all duration-300 ease-out h-[3px] w-full rounded-sm",
```

Comment:
Is there a reason we use h-[3px] instead of just h-3 here? Also, I see that we have a mix of rem and px units (eg, line 72, line 225 etc - I'm totally fine with border width or border radius be px but with width/height/font sizes, perhaps it's better we go with rem? 👀 I understand there might be exceptions.